src/main/java/io/opentracing/contrib/global/GlobalSpan.java

```diff
@@ -0,0 +1,182 @@
+package io.opentracing.contrib.global;
+
+import io.opentracing.NoopSpan;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import nl.talsmasoftware.context.Context;
+import nl.talsmasoftware.context.threadlocal.AbstractThreadLocalContext;
+
+import java.util.Map;
+
+/**
+ * This span delegates all its core methods to another span implementation, providing a global context.
+ *
+ * @author Sjoerd Talsma
+ */
+class GlobalSpan extends AbstractThreadLocalContext<Span> implements Span {
```

Comment:
To have the Tracer global and the span context implicit, the current actual Span itself must also be globally available (think of logging to the active span).